docs/source/development.rst

```diff
@@ -4,63 +4,210 @@
+
+   $ git add .
+   $ git commit -m 'your commit message'
+   $ git push origin your_branch_name
+
+.. note:: If you are working on a branch with someone else, it is always a good
+          idea to do a ``git pull --rebase`` before pushing. This command
+          updates your branch with remote changes and apply your new commits on
+          top of them.
+
+.. warning:: If your commit message contains the string ``[ci skip]`` the
+	     continuous integration tests are not run. However, try not to use
+	     this feature unless you know what you're doing.
+
+
+Creating a pull request
+-----------------------
+
+Whenever you want to add a feature or push a bugfix, you should make a new pull
+request, which can be reviewed and merged by someone else. The typical workflow
+should be as follows:
+
+1. Create a new branch, make your changes and push them to the repository.
+
+2. You should now see the new branch on the Github project page. When you open
    the branch page, click on "Create Pull request" button.
 
-5. When the pull request is created, the tests are run on Travis. You can see
-   the status of the test run on the pull request page. There is also a link to
-   Travis so you can inspect the results of the test run, and make additional
-   changes in order to make the tests successful, if needed.
+3. When the pull request is created, the continuous integration tests are run on
+   Travis. You can see the status of the test run on the pull request
+   page. There is also a link to Travis so you can inspect the results of the
+   test run, and make additional changes in order to make the tests successful,
+   if needed. Additionally to the code quality checking tools, unit and
+   regression tests are run as well.
+
+When you create a pull request, assign one or two people to do the review.
+
+
+Code review and merging
+-----------------------
+
+Your pull requests should always be subject to code review. After you create the
+pull request, select one or two contributors and assign them to make a review.
+
+This phase consists of discussion about the introduced changes, suggestions, and
+another requirements made by the reviewers. Anyone who wants to do a review can
+contribute, the reviewer roles are not considered exclusive.
+
+After all of the reviewers' comments have been addressed and the reviewers
+approved the pull request, the pull request can be merged. It is usually a good
+idea to rebase the code to the recent version of master. Assuming your working
+copy is switched to the **master** branch, do:
 
-6. Simultaneously to the test runs, anyone now can look at the code changes by
-   clicking on the "Files changed" tab and comment on individual lines in the
-   diff, and approve or reject the proposed pull request.
+.. code-block:: bash
 
-7. When all the tests are passing and the pull request is approved, it can be
-   merged.
+   $ git pull --rebase
+   $ git checkout your_branch_name
+   $ git rebase master
```

Comment:
jo aha, to je pravda, ze to neni napred.. zkusim neco vymyslet jak to napsat lip